File: ABC100_199/ABC170/c.cpp
#include <algorithm>
#include <iostream>
#include <vector>
using namespace std;

int main() {
  int X, N;
  cin >> X >> N;
  vector<int> p(N);
  for (int i = 0; i < N; i++) cin >> p[i];
  int m = 10000;
  int ans = 0;
  for (int i = -100; i < 200; i++) {
    if (abs(X - i) < m) {
      bool flag = true;
      for (int j = 0; j < N; j++) {
        if (i == p[j]) flag = false;
      }
      if (!flag) continue;
      ans = i;
      m = abs(X - i);
    } else {
      continue;
    }
  }

  cout << ans << endl;
  return 0;
}

File: ABC100_199/ABC170/c.py
X, N = map(int, input().split())
P = list(map(int, input().split()))
if(N == 0):
    print(X)
    exit(0)

L = []
for i in range(1, 101):
    if(i in P): continue
    else: L.append(i)
m = 1000
for l in L:
    m = min(m, abs(X - l))
ans = []
for l in L:
    if(abs(X - l) == m):
        ans.append(l)
ans.sort()
print(ans[0])
